DSL/Filter/HasChildFilter.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the ONGR package.
+ *
+ * (c) NFQ Technologies UAB <info@nfq.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ONGR\ElasticsearchBundle\DSL\Filter;
+
+use ONGR\ElasticsearchBundle\DSL\BuilderInterface;
+use ONGR\ElasticsearchBundle\DSL\ParametersTrait;
+
+/**
+ * Elasticsearch has_child filter.
+ */
+class HasChildFilter implements BuilderInterface
+{
+    use ParametersTrait;
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var string
+     */
+    private $query;
+
+    /**
+     * @param string           $type
+     * @param BuilderInterface $query
+     * @param array            $parameters
+     */
+    public function __construct($type, BuilderInterface $query, array $parameters = [])
+    {
+        $this->type = $type;
+        $this->query = $query;
+        $this->setParameters($parameters);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getType()
+    {
+        return 'has_child';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toArray()
+    {
+        $query = [
+            'type' => $this->type,
+            'query' => [
```

Comment:
Filter accepts filters and queries, but here it should be `filter` instead of `query`.